plugins/catalog-backend/src/service/createRouter.ts

```diff
@@ -239,71 +288,224 @@ export async function createRouter(
                 prevCursor: encodeCursor(pageInfo.prevCursor),
               }),
             },
-          }),
-        });
+          };
+
+          await auditorEvent?.success({
+            // Let's not log out the entities since this can make the log very big
+            meta,
+          });
+
+          await writeEntitiesResponse({
+            res,
+            items,
+            alwaysUseObjectMode: !disableRelationsCompatibility,
+            responseWrapper: entities => ({
+              items: entities,
+              ...meta,
+            }),
+          });
+        } catch (err) {
+          await auditorEvent?.fail({
+            error: err,
+          });
+          throw err;
+        }
       })
       .get('/entities/by-uid/:uid', async (req, res) => {
         const { uid } = req.params;
-        const { entities } = await entitiesCatalog.entities({
-          filter: basicEntityFilter({ 'metadata.uid': uid }),
-          credentials: await httpAuth.credentials(req),
+
+        const auditorEvent = await auditor?.createEvent({
+          eventId: 'entity-fetch',
+          request: req,
+          meta: {
+            queryType: 'by-uid',
+            uid: uid,
+          },
         });
-        writeSingleEntityResponse(res, entities, `No entity with uid ${uid}`);
+
+        try {
+          const { entities } = await entitiesCatalog.entities({
+            filter: basicEntityFilter({ 'metadata.uid': uid }),
+            credentials: await httpAuth.credentials(req),
+          });
+
+          await auditorEvent?.success({
+            meta: {
+              entities: entities,
+            },
+          });
+
+          writeSingleEntityResponse(res, entities, `No entity with uid ${uid}`);
```

Comment:
```suggestion       
          writeSingleEntityResponse(res, entities, `No entity with uid ${uid}`);
          
          await auditorEvent?.success({
            meta: {
              entities: entities,
            },
          });
```
Should `writeSingleEntityResponse` be called before `auditorEvent` to ensure that the success message isn't logged in case the error "No entity with uid ${uid}" occurs?